docs/ref/models/indexes.txt

```diff
@@ -0,0 +1,75 @@
+=====================
+Model index reference
+=====================
+
+.. module:: django.db.models.indexes
+   :synopsis: Built-in index types.
+
+.. currentmodule:: django.db.models
+
+.. versionadded:: 1.11
+
+Index classes ease creating database indexes. This document explains the API
+references of :class:`Index` This includes the `index options`_ and the
+`built-in indexes`_ that Django offers.
+
+.. note::
+
+    Indexes are defined in ``django.db.models.indexes``, but for convenience
+    they're imported into :mod:`django.db.models`. The standard convention is
+    to use ``from django.db import models`` and refer to the indexes as
+    ``models.<IndexClass>``.
+
+``Index`` options
+=================
+
+.. class:: Index(fields, name)
```

Comment:
I guess we shouldn't have `.. class:: Index` twice though (it also appears under "Built-in" indexes) since that will be ambiguous from a linking perspective.